kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -188,4 +212,56 @@ private void waitRandomTime() throws InterruptedException {
     // wait
     Thread.sleep(retryWaitMs + random.nextInt(WAIT_MAX_JITTER));
   }
+
+  /**
+   * Filter and send Records to DLQ according to allowedBigQueryReason List
+   * @param rows The rows to write.
+   * @param failedRowsMap map from failed row id to the BigQueryError
+   * @param table The BigQuery table to write the rows to
+   * @return Filtered Map from failed row id to the BigQueryError which cannot be sent to DLQ
+   */
+  private Map<Long, List<BigQueryError>> filterAndSendRecordsToDLQ(
+      SortedMap<SinkRecord, InsertAllRequest.RowToInsert> rows,
+      Map<Long, List<BigQueryError>> failedRowsMap,
+      PartitionedTableId table
+  ) {
+    long index = 0;
+    Set<SinkRecord> recordsToDLQ = new TreeSet<>(rows.comparator());
+    Map<Long, List<BigQueryError>> recordsToDLQFailureMap = new TreeMap<>();
+    Map<Long, List<BigQueryError>> updatedFailedRowMap = new TreeMap<>();
+    for (Map.Entry<SinkRecord, InsertAllRequest.RowToInsert> row : rows.entrySet()) {
+      if (failedRowsMap.containsKey(index)) {
+        if (errantRecordHandler.isErrorReasonAllowed(failedRowsMap.get(index).get(0).getReason())) {
+          recordsToDLQ.add(row.getKey());
+          recordsToDLQFailureMap.put(index, failedRowsMap.get(index));
+        } else {
+          updatedFailedRowMap.put(index, failedRowsMap.get(index));
+        }
+      }
+      index++;
+    }
+
+    if (errantRecordHandler.getErrantRecordReporter() != null) {
```

Comment:
nit: This should not be needed as we already have a check.

Reply:
This is like a fail-safe option incase someone forgets to add this condition in future cases where this function is called. I think this is a harmless change. Let me know if it makes sense to keep it otherwise will remove.

Reply:
yup, we can keep.